pkgs/development/tools/xcbuild/platform.nix

```diff
@@ -188,6 +188,22 @@ let
       Type = "ProductType";
       Name = "Dynamic Library";
       PackageTypes = [ "com.apple.package-type.mach-o-dylib" ];
+      DefaultBuildProperties = {
```

Comment:
Without this it seems to try to compile my libutil dylib as an executable and complains about a missing `_main` symbol.